mbuild/lib/recipes/polymer.py

```diff
@@ -45,40 +148,241 @@ def __init__(self, monomers, n, sequence="A", port_labels=("up", "down")):
             )
 
         # 'A': monomer_1, 'B': monomer_2....
-        seq_map = dict(zip(unique_seq_ids, monomers))
-
+        seq_map = dict(zip(unique_seq_ids, self._monomers))
         last_part = None
         for n_added, seq_item in enumerate(it.cycle(sequence)):
             this_part = clone(seq_map[seq_item])
             self.add(this_part, "monomer[$]")
             if last_part is None:
                 first_part = this_part
             else:
```

Comment:
`first_part` is never used. I think we can simplify this logic to:
```suggestion
            if last_part is not None:
```